packages/record-data/addon/-private/record-data.ts

```diff
@@ -465,23 +465,47 @@ export default class RecordDataDefault implements RelationshipRecordData {
   }
 
   /**
-    Computes the set of internal models reachable from `this` across exactly one
+    Iterates over the set of internal models reachable from `this` across exactly one
     relationship.
-
-    @return {Array} An array containing the internal models that `this` belongs
-    to or has many.
-
   */
-  _directlyRelatedRecordDatas(): RecordData[] {
-    let array = [];
-
-    this._relationships.forEach((name, rel) => {
-      let members = rel.members.list;
-      let canonicalMembers = rel.canonicalMembers.list;
-      array = array.concat(members, canonicalMembers);
-    });
-    return array;
-  }
+  _directlyRelatedRecordDatasIterable = () => {
+    const relationships = Object.values(this._relationships.initializedRelationships) as Relationship[];
+
+    let i = 0;
+    let j = 0;
+    let k = 0;
+    let nextMember;
+
+    const findNext = () => {
+      while (i < relationships.length) {
+        while (j < 2) {
+          let members = j === 0 ? relationships[i].members.list : relationships[i].canonicalMembers.list;
+          while (k < members.length) {
+            return members[k++];
+          }
+          k = 0;
+          j++;
+        }
+        j = 0;
+        i++;
+      }
+      return undefined;
+    };
+
+    nextMember = findNext();
+
+    return {
+      [Symbol.iterator]() {
```

Comment:
`Symbol` is not supported in IE

Reply:
@btecu 
I understand that IE does not support `Symbol.iterator`, but running our application with this modification works perfectly with IE11. So, I assume that babel is taking care of this.
For the iterator in belongs-to-test, I can easily drop the stub function `_directlyRelatedRecordDatasIterable` and directly stub `_allRelatedRecordDatas` to return just `[this]`.
Or am I missing something?